### PR TITLE
Integrated retrofit for Destinations module and in CityFragment

### PR DESCRIPTION
**Pull request by Sumeet-Rukeja**

### CODE REVIEW

**Comment by Sumeet-Rukeja** [comment]
@gugakatsi Actually, my intention is to make a common retrofit module for all the api modules. That's why i have used common naming convention. Let me know if i should change my approach.

File: Android/app/build.gradle [hunk 1]
```diff
@@ -36,7 +36,7 @@ dependencies {
     compile fileTree(include: ['*.jar'], dir: 'libs')
     testCompile 'junit:junit:4.12'
 
-    compile 'com.squareup.okhttp3:okhttp:3.4.1'
+    compile 'com.squareup.okhttp3:okhttp:3.9.1'
```
Comment (Swati4star) [inline Android/app/build.gradle]: Is there a need to upgrade the version?

File: Android/app/src/main/java/data/ApiInterface.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ * Created by Sumeet on 12/17/2017.
+ */
+
+public interface ApiInterface {
+
+    @GET(Constants.ALL_CITY)
+    Call<AllCities> callAllCitiesApi();
+
+    @GET(Constants.CITY_INFO)
+    Call<CityInfoModel> callCityInfoApi(@Query("id") String id);
```
Comment (Swati4star) [inline Android/app/src/main/java/data/ApiInterface.java]: We can move these strings "id", "query" to some constants file

File: Android/app/src/main/java/data/models/AutocompleteModel/Autocomplete.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+
```
Comment (Swati4star) [inline Android/app/src/main/java/data/models/AutocompleteModel/Autocomplete.java]: extra line space

File: Android/app/src/main/java/data/models/CityInfoModel/CityInfoModel.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
+
```
Comment (Swati4star) [inline Android/app/src/main/java/data/models/CityInfoModel/CityInfoModel.java]: extra line space

File: Android/app/src/main/java/data/models/CityInfoModel/Weather.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
+
```
Comment (Swati4star) [inline Android/app/src/main/java/data/models/CityInfoModel/Weather.java]: same here

File: Android/app/src/main/java/tie/hackathon/travelguide/CityFragment.java [hunk 6]
```diff
@@ -48,31 +52,42 @@
...
-    public CityFragment() {}
+    @BindView(R.id.cityname)
+    AutoCompleteTextView cityname;
+    @BindView(R.id.pb)
+    ProgressBar pb;
+    @BindView(R.id.music_list)
+    ListView lv;
+
+    List<String> id = new ArrayList<>();
+    List<String> list2 = new ArrayList<>();
```
Comment (Swati4star) [inline Android/app/src/main/java/tie/hackathon/travelguide/CityFragment.java]: change the name from list2 to something relevant

File: Android/app/src/main/java/tie/hackathon/travelguide/CityFragment.java [hunk 7]
```diff
@@ -48,31 +52,42 @@
...
-    private Typeface    tex;
-    private Handler     mHandler;
-
-    public CityFragment() {}
+    @BindView(R.id.cityname)
+    AutoCompleteTextView cityname;
+    @BindView(R.id.pb)
+    ProgressBar pb;
+    @BindView(R.id.music_list)
+    ListView lv;
```
Comment (Swati4star) [inline Android/app/src/main/java/tie/hackathon/travelguide/CityFragment.java]: change variable name lv

File: Android/app/src/main/java/tie/hackathon/travelguide/CityFragment.java [hunk 8]
```diff
@@ -98,186 +133,143 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
+            public void onFailure(retrofit2.Call<List<Autocomplete>> call, Throwable t) {
+                GlobalClass.showApiFailureMessage(getActivity());
             }
-        });
-    }
+    };
 
-    private void getCity() {
+    private void setAutocompleteAdapter() {
+        final ArrayList list, list1;
```
Comment (Swati4star) [inline Android/app/src/main/java/tie/hackathon/travelguide/CityFragment.java]: name the variables something relevant

File: Android/app/src/main/java/tie/hackathon/travelguide/destinations/funfacts/FunFacts.java [hunk 9]
```diff
@@ -10,6 +10,7 @@
 import android.support.v4.view.ViewPager;
 import android.support.v7.app.AppCompatActivity;
 import android.support.v7.widget.Toolbar;
+import android.widget.Toast;
```
Comment (Swati4star) [inline Android/app/src/main/java/tie/hackathon/travelguide/destinations/funfacts/FunFacts.java]: where is the need for toast?

File: Android/app/src/main/java/tie/hackathon/travelguide/destinations/funfacts/FunFactsPresenter.java [hunk 10]
```diff
@@ -1,29 +1,41 @@
 package tie.hackathon.travelguide.destinations.funfacts;
 
 import android.util.Log;
+import android.widget.Toast;
```
Comment (Swati4star) [inline Android/app/src/main/java/tie/hackathon/travelguide/destinations/funfacts/FunFactsPresenter.java]: no need of toast

**Comment by Swati4star** [comment]
I have reviewed the code & added a few comments.

**Comment by gugakatsi** [comment]
@Swati4star i think each module should have each own apiInterface , and named in independently for example TravelModuleApiInterface something like that , i don't think it's good idea to shrink those in one ApiInterface, what do u think about that.

File: Android/app/build.gradle (hunk 1 above)
Comment (Sumeet-Rukeja) [reply Android/app/build.gradle]: Not necessary to update it. I thought it'll be better if we keep 3rd party libraries up-to-date.

**Comment by Sumeet-Rukeja** [comment]
Hi @Swati4star , i have made changes according to your review. As far as extra lines issue is concerned, they are auto generated when we reformat the code. Please let me know if there are anymore changes for this module.

**Comment by Swati4star** [comment]
Hi @Sumeet-Rukeja, the modules have been moved now. Add the changes to resolve the conflicts.

Or, it might be better to create a new PR. Closing this one for now. Feel free to open another one, if you start working on it again!

**Pull request closed without merging**